app/forms/flood_risk_engine/steps/correspondence_contact_email_form.rb

```diff
@@ -0,0 +1,73 @@
+require "validates_email_format_of"
+require "active_model/validations/confirmation"
+
+module FloodRiskEngine
+  module Steps
+    class CorrespondenceContactEmailForm < BaseForm
+
+      def self.factory(enrollment)
+        enrollment.correspondence_contact ||= FloodRiskEngine::Contact.new(contact_type: :correspondence)
+
+        new(enrollment.correspondence_contact, enrollment)
+      end
+
+      def self.params_key
+        :correspondence_contact_email
+      end
+
+      property :email_address
+      property :email_address_confirmation, virtual: true
+
+
+      # This group manages the main email_address field, confirmation is dependent on results of this group
+      validation :email_address_valid do
+
+        validates :email_address, presence: {
+          message: I18n.t("#{CorrespondenceContactEmailForm.locale_key}.errors.email_address.blank")
+        }
+
+        validates :email_address, email_format: {
+          allow_blank: true,
+          message: I18n.t("#{CorrespondenceContactEmailForm.locale_key}.errors.email_address.format")
+        }
+
+      end
+
+      # Could not get any 'if' forms for example -  if: changed?(:email_address) - to work that would monitor
+      # email address, so this reform validation group only runs, if previous email group passed,
+      # which achieves the same result
+      #
+      validation :confirmation_present, if: :email_address_valid do
+        validates :email_address_confirmation, presence: {
+          message: I18n.t("#{CorrespondenceContactEmailForm.locale_key}.errors.email_address_confirmation.blank")
+        }
+      end
+
+      # The helpers ; allow_blank: true, allow_nil: true  ; only seem to apply to the email_address field
+      # not the confirmation, so this format prevents both format & blank messages when conf = blank
+      #
+      validation :passwords_match, if: :confirmation_present do
+        # Although not specified, Rails automatically validates against a field called email_address_confirmation
+        validates :email_address, confirmation: {
+          message: I18n.t("#{CorrespondenceContactEmailForm.locale_key}.errors.email_address_confirmation.format")
+        }
+      end
+
+
+      def save
+        super
+        enrollment.correspondence_contact ||= model
+        enrollment.save
+      end
+
+      # Force use of the factory to create instances of this class
+      class << self
+        private
+        def new(model, enrollment = nil)
+          super(model, enrollment)
+        end
+      end
+
+    end
+  end
+end
```

Comment:
good to know how to reach the form. I don't think its as readable though.

helper is nice idea will add